nautobot/extras/models/customfields.py

```diff
@@ -168,6 +164,16 @@ def remove_stale_data(self, content_types):
     def clean(self):
         super().clean()
 
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+
+            if self.name != database_object.name:
+                raise ValidationError({"name": "Name cannot be changed once created"})
+
+            if self.type != database_object.type:
+                raise ValidationError({"type": "Type cannot be changed once created"})
```

Comment:
Seems to me we might also make set `editable=False` on the `name` and `type` fields to coincide with this validation?

That would simplify the edit views perhaps?

Reply:
My understanding of editable is it just excludes the field from inclusion in forms and model validation. In the admin pannel we could implement `get_readonly_field()` but I would rather not, and focus on a solution once we move away from the admin pannel.

Reply:
Works for me.